test/AngularSpec.js

```diff
@@ -1683,6 +1683,38 @@ describe('angular', function() {
       dealoc(appElement);
     });
 
+    it('should bootstrap from an extension into an extension document for same-origin documents only', function() {
+      // Extension URLs are browser-specific, so we must choose a scheme that is supported by the browser to make
+      // sure that the URL is properly parsed.
+      var extensionScheme;
+      var userAgent = window.navigator.userAgent;
+      if (/Firefox\//.test(userAgent)) {
+        extensionScheme = 'moz-extension';
+      } else if (/Edge\//.test(userAgent)) {
+        extensionScheme = 'ms-browser-extension';
+      } else if (/Chrome\//.test(userAgent)) {
+        extensionScheme = 'chrome-extension';
+      } else if (/Safari\//.test(userAgent)) {
+        extensionScheme = 'safari-extension';
+      } else if (/Trident/.test(userAgent)) {
+        return;  // IE does not support extensions with a standard protocol, so skip the test.
+      } else {
```

Comment:
I see. In that case this test shouldn't be run in IE at all as it will return `true` immediately in the very first `document.currentScript` check as it doesn't support this API. Maybe add:
```js
if (msie) return true;
```
at the beginning of the test you added?